lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,24 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n*k), where n is the size of the external array that represents
+# the rows and k is the size of an individual array that represents the columns
+# Space complexity: O(1), because the additional space required is two variables to store
+# the sums for rows and columns and that doesn't depend on the size of the matrix
 def matrix_check_sum(matrix)
-  raise NotImplementedError
+  rows = matrix.length
+  columns = matrix[0].length
+
+  rows.times do |r|
+    row_sum = 0
+    column_sum = 0
+    columns.times do |c|
+      row_sum += matrix[r][c]
+      column_sum += matrix[c][r]
+    end
+    return row_sum == column_sum
```

Comment:
This is causing your method to only evaluate the *first* row and column.  Here's some input that should return `false` but is returning `true`:
```ruby
[1] pry(main)> matrix = [[0, 0, 2], [2, 0, 0], [0, 0, 0]]
=> [[0, 0, 2], [2, 0, 0], [0, 0, 0]]
[2] pry(main)> matrix_check_sum matrix
=> true
```
The sum of the second row is 2, but the sum of the second column is 0, so this shouldn't pass the check.